api-metastore/src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -399,6 +399,7 @@ public void delete(final String eventTypeName) throws EventTypeDeletionException
     private Multimap<TopicRepository, String> deleteEventTypeWithSubscriptions(final String eventType) {
         try {
             return transactionTemplate.execute(action -> {
+                eventTypeRepository.lockTable(EventTypeRepository.TableLock.ROW_EXCLUSIVE);
```

Comment:
I suggest we change it to something like the following:
```
eventTypeRepository.lockingTable(transactionTemplate, EventTypeRepository.TableLock.ROW_EXCLUSIVE,
    action -> {
        ...
    });
```

Otherwise we allow the caller to lock the table and walk away forgetting to unlock (by closing the transaction).